android/src/main/java/dev/openfeature/sdk/OpenFeatureAPI.kt

```diff
@@ -11,7 +11,11 @@ object OpenFeatureAPI {
     fun setProvider(provider: FeatureProvider, initialContext: EvaluationContext? = null) {
         this@OpenFeatureAPI.provider = provider
         if (initialContext != null) context = initialContext
-        provider.initialize(context)
+        try {
+            provider.initialize(context)
+        } catch (e: Throwable) {
+            // This should never happen
```

Comment:
Can we rephrase this to 
> This is not allowed to happen.

and maybe reference the spec?

Reply:
I don't think there is anything about this in the Spec. It's this specific implementation that expects the Provider to never throw and rather reads ProviderError events from the Provider.
The Java SDK, for example, works in the opposite way and expects the Provider to throw [in case of errors](https://github.com/open-feature/java-sdk/blob/1b6e16017783aa45ece0c7a6b5ba0f01100813c9/src/main/java/dev/openfeature/sdk/ProviderRepository.java#L140C16-L140C16)